rosetta/transactions/intent.go

```diff
@@ -0,0 +1,174 @@
+
+	firstOpNegative := strings.HasPrefix(operations[0].Amount.Value, "-")
+	secondOpNegative := strings.HasPrefix(operations[1].Amount.Value, "-")
+
+	// Operations are invalid if both operations have negative amounts.
+	if firstOpNegative && secondOpNegative {
+
+		return nil, failure.InvalidIntent{
+			Description: failure.NewDescription("invalid operations - two deposits specified"),
+			Sender:      operations[0].AccountID.Address,
+			Receiver:    operations[1].AccountID.Address,
+		}
+	}
+
+	// Operations are also invalid if both operations have positive amounts.
+	if !firstOpNegative && !secondOpNegative {
+
+		return nil, failure.InvalidIntent{
+			Description: failure.NewDescription("invalid operations - two withdrawals specified"),
+			Sender:      operations[0].AccountID.Address,
+			Receiver:    operations[1].AccountID.Address,
+		}
+	}
+
+	// Assume the first operation is the one with the negative amount.
+	send := operations[0]
+	receive := operations[1]
+
+	// If that was not the case, switch the send and receive operations.
+	if !strings.HasPrefix(send.Amount.Value, "-") {
+		receive = operations[0]
+		send = operations[1]
+	}
+
+	// Validate the sender and the receiver account IDs.
+	err := p.validate.Account(send.AccountID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid sender account: %w", err)
+	}
+	err = p.validate.Account(receive.AccountID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid receiver account: %w", err)
+	}
+
+	// Validate the currencies specified for deposit and withdrawal.
+	send.Amount.Currency, err = p.validate.Currency(send.Amount.Currency)
+	if err != nil {
+		return nil, fmt.Errorf("invalid sender currency: %w", err)
+	}
+	receive.Amount.Currency, err = p.validate.Currency(receive.Amount.Currency)
+	if err != nil {
+		return nil, fmt.Errorf("invalid receiver currency: %w", err)
+	}
+
+	// Make sure that both the send and receive operations use the same currency.
+	// This is perhaps unnecessary at the moment since we only have a single currency.
+	if send.Amount.Currency != receive.Amount.Currency {
+		return nil, failure.InvalidIntent{
+			Sender:      send.AccountID.Address,
+			Receiver:    receive.AccountID.Address,
+			Description: failure.NewDescription("send and receive currencies do not match"),
+		}
+	}
+
+	// Parse value specified by the sender, after removing the negative sign prefix.
+	trimmed := strings.TrimPrefix(send.Amount.Value, "-")
+	sv, err := convert.ParseRosettaValue(trimmed)
+	if err != nil {
+		return nil, failure.InvalidIntent{
+			Sender:   send.AccountID.Address,
+			Receiver: receive.AccountID.Address,
+			Description: failure.NewDescription("could not parse withdrawal amount",
+				failure.WithString("withdrawal_amount", send.Amount.Value),
+				failure.WithErr(err),
+			),
+		}
+	}
+	// Parse value specified by the receiver.
+	rv, err := convert.ParseRosettaValue(receive.Amount.Value)
+	if err != nil {
```

Comment:
To be consistent with the rest of the func there should be a newline before this new block

```suggestion
	}
	
	// Parse value specified by the receiver.
	rv, err := convert.ParseRosettaValue(receive.Amount.Value)
	if err != nil {
```